### PR TITLE
Add layers to image-config history

### PR DESCRIPTION
**Pull request by guyboltonking**
This keeps the config history consistent with the layers in the manifest.  It also:

* adds the bazel label for the layer as an annotation to the layer descriptor, and a comment to the layer history entry;
* passes through the descriptor created by oci_image_layer to oci_image (previously it was discarded and recreated from scratch);
* updates ocispec so we can use all the current annotations in that package.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,6 +91,12 @@ dependencies.
 
 ### Developing
 
+#### Updating dependencies
+
+Run `bazel run :go -- get -u DEPENDENCY && bazel run :gazelle-update-repos`
```
Comment (jprobinson) [inline README.md:96]: I think we may want to remove the `-u` here. That tells the toolchain to grab the dependency but also upgrade any related deps, which may not always be the expected result.

**Review by jprobinson — APPROVED** [review]
🧹 🌮

**Comment by guyboltonking** [comment]
@guyboltonking queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/DataDog/rules_oci/37).

File: cmd/ocitool/appendlayer_cmd.go [hunk 2]
```diff
@@ -129,23 +129,21 @@ func AppendLayersCmd(c *cli.Context) error {
...
+	layerPaths := c.Generic("layer").(*flagutil.KeyValueFlag).Map
 
 	layerProvider := &blob.Index{
 		Blobs: make(map[digest.Digest]string),
 	}
 
 	layerDescs := make([]ocispec.Descriptor, 0, len(layerPaths))
-	for _, lp := range layerPaths {
-		ld, reader, err := ociutil.CreateDescriptorFromFile(lp)
+	for layerPath, layerDescriptorPath := range layerPaths {
```
Comment (guyboltonking) [inline cmd/ocitool/appendlayer_cmd.go:139]: OOPS.  Non-deterministic map iteration means the layers we carefully specify in order on the command line end up all topsy-turvy.  Fix incoming.

**Pull request merged**